srcs/utils/error_utils.c

```diff
@@ -0,0 +1,21 @@
+#include "minishell.h"
+
+int		error_and_exit(t_error code)
+{
+	t_minishell *minishell;
+
+	minishell = get_minishell(NULL);
+	error_message(code); // later on, maybe add the program_pathname stored in struct to print a clearer error
+	if (minishell)
+		free_minishell(minishell);
+	exit(EXIT_FAILURE);
+}
+
+void	error_message(t_error code)
+{
+	if (code == ERR_NO_PRINT)
+		return ;
+	// we could print the program name here
+	if (MEMORY_FAIL) //in the meantime before linking libft. 
+		printf("%s\n", strerror(errno)); // don't forget to print on STDERR_FILENO

```

Comment:
Where is this variable errno comming from? And if you want to print on stderr (which I support 100%), then we should use write instead of printf

Reply:
From what I understand, errno will pull the right error value given by your system if it's an error that is already coded in. So Memory failure for example is pretty classic. We'll need to come up with our own error message for errors that aren't in there. 
If I understood right. 

And yeah, of course, it's just that I would use ft_putstr_fd and we haven't linked the libft yet. It was just to check if the structure worked.

Reply:
Understood for the printf vs write part, but I still don't get where errno is comming from. I mean, we're not defining it/assigning it anywhere

Reply:
Aaaah ! It's in the <errno.h> header I added